Model/ModelManager.php

```diff
@@ -376,7 +386,7 @@ public function getNormalizedIdentifier($entity)
         }
 
         // the entities is not managed
-        if (!$entity /*|| !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity) // commented for perfomance concern */) {
+        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
```

Comment:
This check is needed because I set the uuid in php code (in the constructor) but that way sonata thinks it's an existing entity and goes to the edit screen, but it should check doctrine if the entity exists and decised that way if it is a create or an update.

I will check when and why this was added and try to get a benchmark!